### DISCUSSION TITLE
DNS-over-QUIC support

### DISCUSSION THREAD

**Discussion by Ehab-d** [DNSCrypt/dnscrypt-proxy#2496] [Ideas]
Adding support for DNS-over-QUIC to dnscrypt-proxy.

**Comment by jedisct1** [comment]
DNS can use any transport protocol, and it's easy to come with a new DNS-over-something, but the result is not always very useful in practice.

There's already DNSCrypt that mostly uses UDP. And DoH, that already uses QUIC with HTTP/3 that `dnscrypt-proxy` supports. DoH was made to leverage existing web servers, HTTP caches and CDNs. 
Most importantly, both protocols can hide client IP addresses (Anonymized DNSCrypt, ODoH, DoOH).

Doing just DNS over QUIC without HTTP doesn't have any of these properties. It's trivial to block, can't be cached using HTTP caches, can't anonymize client IPs, yet requires a ginormous amount of code and thus possible vulnerabilities compared to DNSCrypt. 

I'd rather wait for the next DNS-over-something. In the meantime, DoOH is more likely to catch up, so this is the one that will be implemented next.

**Reply by Ehab-d** [reply]
However, I would argue that DNS-over-QUIC without HTTP/3 still has some potential benefits. For example, it could be faster than DNSCrypt or DoH.

**Reply by jedisct1** [reply]
Not really.

**Comment by ghost** [comment]
I think cookies and overhead are main disadvantages of DoH/3, but the most concerning aspect of DoQ is one described as "can be tied to client device". How is that? Is it because DoQ is rarely used or is it something in packets that gives away more information than DoT does?

**Comment by c-michail** [comment]
+1 for DNS Quic. Adguard DNS Proxy supports it and it's working great!

**Reply by serhii-nakon** [reply]
+1 it useful
